From akpm@linux-foundation.org  Wed Nov 12 11:46:12 2008
From: Steven Noonan <steven@uplinklabs.net>
Date: Wed, 05 Nov 2008 12:41:24 -0800
Subject: USB: EHCI pci-quirks.c: don't wait so long for BIOS handoff
To: mm-commits@vger.kernel.org
Cc: steven@uplinklabs.net, david-b@pacbell.net, greg@kroah.com, jeff@garzik.org, stern@rowland.harvard.edu
Message-ID: <200811052041.mA5KfOtE011772@imap1.linux-foundation.org>

From: Steven Noonan <steven@uplinklabs.net>

Instead of waiting a painful 5000ms, quirk_usb_disable_ehci() now does a
1000ms loop to wait for the BIOS to acknowledge the handoff.

The five second delay is really quite irritating to have to deal with
every boot up, and I very seriously doubt any non-broken bios takes more
than a second to do the actual handoff.

Signed-off-by: Steven Noonan <steven@uplinklabs.net>
Cc: Alan Stern <stern@rowland.harvard.edu>
Cc: David Brownell <david-b@pacbell.net>
Cc: Jeff Garzik <jeff@garzik.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/pci-quirks.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/host/pci-quirks.c
+++ b/drivers/usb/host/pci-quirks.c
@@ -271,7 +271,7 @@ static void __devinit quirk_usb_disable_
 			/* if boot firmware now owns EHCI, spin till
 			 * it hands it over.
 			 */
-			msec = 5000;
+			msec = 1000;
 			while ((cap & EHCI_USBLEGSUP_BIOS) && (msec > 0)) {
 				tried_handoff = 1;
 				msleep(10);
